dfpl/options.py

```diff
@@ -132,39 +99,116 @@ class GnnOptions(TrainArgs):
     preds_path: str = "./tox21dmpnn.csv"
     test_path: str = ""
     save_preds: bool = True
+    calibration_method: str = "none"
+    uncertainty_method: str = "none"
+    calibration_path: str = ""
+    evaluation_methods: str = "none"
+    evaluation_scores_path: str = ""
+    wabTracking: bool = False
+    split_sizes: List[float] = None
+
+    # save_smiles_splits: bool = False
 
     @classmethod
-    def fromCmdArgs(cls, args: argparse.Namespace) -> GnnOptions:
-        """
-        Creates Options instance from cmdline arguments.
+    def fromCmdArgs(cls, args: argparse.Namespace, json_config: Optional[dict] = None):
+        # Initialize with JSON config if provided
+        if json_config:
+            opts = cls(**json_config)
+        else:
+            opts = cls()
 
-        If a training file (JSON) is provided, the values from that file are used.
-        However, additional commandline arguments will be preferred. If, e.g., "fpSize" is specified both in the
-        JSON file and on the commandline, then the value of the commandline argument will be used.
-        """
-        result = GnnOptions()
-        if "configFile" in vars(args).keys():
-            jsonFile = Path(makePathAbsolute(args.configFile))
-            if jsonFile.exists() and jsonFile.is_file():
-                with jsonFile.open() as f:
-                    content = f.read()
-                    result = jsonpickle.decode(content)
-            else:
-                raise ValueError("Could not find JSON input file")
-
-        return result
+        # Update with command-line arguments
+        for key, value in vars(args).items():
+            if value is not None:
+                setattr(opts, key, value)
+
+        return opts
+
+
+class PredictGnnOptions(PredictArgs):
+    """
+    Dataclass to hold all options used for training the graph models
+    """
+
+    configFile: str = "./example/predictgnn.json"
+    calibration_atom_descriptors_path: str = None
+    calibration_features_path: str = None
+    calibration_interval_percentile: float = 95
+    calibration_method: Literal[
+        "zscaling",
+        "tscaling",
+        "zelikman_interval",
+        "mve_weighting",
+        "platt",
+        "isotonic",
+    ] = None
+    calibration_path: str = None
+    calibration_phase_features_path: str = None
+    drop_extra_columns: bool = False
+    dropout_sampling_size: int = 10
+    evaluation_methods: List[str] = None
+    evaluation_scores_path: str = None
+    # no_features_scaling: bool = True
+    individual_ensemble_predictions: bool = False
+    preds_path: str = None
+    regression_calibrator_metric: Literal["stdev", "interval"] = None
+    test_path: str = None
+    uncertainty_dropout_p: float = 0.1
+    uncertainty_method: Literal[
```

Comment:
Btw: To be correct this needs to be `Optional[Literal[...]]`. Otherwise you can't assign `None`